src/connectionController.ts

```diff
@@ -204,19 +225,233 @@ export default class ConnectionController {
     }
   }
 
+  _notifyKeytarMigrationError(migrationError: Error) {
+    const notificationAction = {
+      title: "Don't show again",
+    };
+
+    const keytarUnavailableErrAlreadyShown = this._storageController.get(
+      StorageVariables.GLOBAL_KEYTAR_ERR_KEYTAR_UNAVAILABLE_NOTIFIED
+    );
+    if (
+      migrationError instanceof KeytarModuleUnavailableError &&
+      !keytarUnavailableErrAlreadyShown
+    ) {
+      void vscode.window
+        .showInformationMessage(
+          `Failed to update connection storage - ${migrationError.message}`,
+          notificationAction
+        )
+        .then((action) => {
+          if (action === notificationAction) {
```

Comment:
What is the purpose here to check the type of action? Then handles only the response of showInformationMessage it is attached to.

Reply:
yea that's right. We probably are fine without that check as well, given that we only have one action. In the original write up I had two actions here and I just did not remove this check after I got rid of the other action.